packages/cli/src/lib/implementation/core-config.middleware.unit.test.ts

```diff
@@ -0,0 +1,58 @@
+import { describe, expect, vi } from 'vitest';
+import { ConfigPathError } from '@code-pushup/core';
+import { CoreConfig } from '@code-pushup/models';
+import { CORE_CONFIG_MOCK } from '@code-pushup/testing-utils';
+import { coreConfigMiddleware } from './core-config.middleware';
+
+vi.mock('@code-pushup/core', async () => {
+  const core: object = await vi.importActual('@code-pushup/core');
+  return {
+    ...core,
+    readRcByPath: vi.fn().mockImplementation((filepath: string): CoreConfig => {
+      const extension = filepath.split('.');
+      if (filepath.includes('throw-error')) {
+        throw new ConfigPathError(filepath);
+      }
+      return {
+        ...CORE_CONFIG_MOCK,
+        upload: {
+          ...CORE_CONFIG_MOCK.upload,
+          project: `cli-${extension}`,
+        },
+      };
+    }),
+    autoloadRc: vi.fn().mockImplementation(
+      (): CoreConfig => ({
+        ...CORE_CONFIG_MOCK,
+        upload: {
+          ...CORE_CONFIG_MOCK.upload,
+          project: `cli-autoload`,
+        },
+      }),
+    ),
+  };
+});
+
+describe('coreConfigMiddleware', () => {
+  it('should load code-pushup.config.(ts|mjs|js) by default', async () => {
+    const config = await coreConfigMiddleware({});
+    expect(config?.upload?.project).toBe('cli-autoload');
+  });
+
+  it.each(['ts', 'mjs', 'js'])(
```

Comment:
This original set of functions should go back to the integration test of `readRcByPath` (which currently only tests `.js` for testing happy paths for all supported files. Having it as a unit test makes us lose coverage and certainty that all types are correctly loaded. All config files are already prepared in `testing-utils` for this.

Reply:
All that is needed is to un-delete `core-config.middleware.integration.test.ts`, right?

Reply:
undeleted